src/components/text-editor/link-menu/limel-text-editor-link-menu.tsx

```diff
@@ -0,0 +1,150 @@
+ * It allows the user to input a title of a link and a link itself.
+ */
+
+@Component({
+    tag: 'limel-text-editor-link-menu',
+    shadow: true,
+    styleUrl: 'limel-text-editor-link-menu.scss',
+})
+export class TextEditorLinkMenu {
+    /**
+     * The link
+     */
+    @Prop()
+    public link: EditorTextLink;
+
+    /**
+     * Defines the language for translations.
+     * Will translate the translatable strings on the components.
+     */
+    @Prop({ reflect: true })
+    public language: Languages = 'en';
+
+    /**
+     * Open state of the dialog
+     */
+    @Prop()
+    public isOpen: boolean = false;
+
+    @State()
+    private invalidLink: boolean = false;
+
+    /**
+     * Emitted when the menu is closed from inside the component.
+     * (*Not* emitted when the consumer sets the `open`-property to `false`.)
+     */
+    @Event()
+    private cancel: EventEmitter<void>;
+
+    /**
+     * Emitted when the menu is saved from inside the component.
+     */
+    @Event()
+    private save: EventEmitter<void>;
+
+    /**
+     * Emitted when the user inputs new values for the link
+     */
+    @Event()
+    private linkChange: EventEmitter<EditorTextLink>;
+
+    public render() {
+        return [
+            <limel-input-field
+                label="Text"
+                value={this.link?.text || ''}
+                leadingIcon="text_cursor"
+                onChange={this.handleLinkTitleChange}
+            />,
+            <limel-input-field
+                label="Link"
+                value={this.link?.href || ''}
+                leadingIcon="-lime-text-link"
+                invalid={this.invalidLink}
+                onChange={this.handleLinkValueChange}
+            />,
+            <div class="actions">
+                <limel-button
+                    label={this.getTranslation('cancel')}
+                    onClick={this.handleCancel}
+                    slot="button"
+                />
+                <limel-button
+                    primary={true}
+                    label={this.getTranslation('save')}
+                    disabled={!this.link?.href || this.invalidLink}
+                    onClick={this.handleSave}
+                    slot="button"
+                />
+            </div>,
+        ];
+    }
+
+    private getTranslation(key: string) {
+        return translate.get(`${key}`, this.language);
+    }
+
+    private validateLink = (href: string): boolean => {
+        if (this.isValidLink(href)) {
+            return true;
+        }
+
+        // If the link doesn't have a scheme, add 'http://' and validate again
+        if (!/^[a-zA-Z][a-zA-Z\d+\-.]*:/.test(href)) {
+            const formattedLink = 'http://' + href;
+            if (this.isValidLink(formattedLink)) {
+                this.link = { ...this.link, href: formattedLink };
+
+                return true;
+            }
+        }
+
+        return false;
+    };
```

Comment:
We might want to put this in https://github.com/Lundalogik/lime-elements/blob/main/src/util/link-helper.ts?rgh-link-date=2024-05-29T13%3A15%3A22Z too, so it can be reused. The stuff that's already there doesn't allow links that are missing the protocol, but even if we don't change the existing behaviour of what's there, we could add this alongside, so it can be reused elsewhere too.

I guess we might want to wait until we're done with the text-editor MVP though.

Reply:
From what I could tell, this doesn't really do anything that the link-helper functions don't already do. I've used the functions in link-helper now to validate the urls. I think if we end up needing further functionality we can add that when needed.